pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -202,19 +203,41 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
 
 		name := fmt.Sprintf("origins-%d", i)
 		providers[name] = provider
-		requirementOrList = append(requirementOrList, &envoy_jwt.JwtRequirement{
-			RequiresType: &envoy_jwt.JwtRequirement_ProviderName{
-				ProviderName: name,
+		requirementAndList = append(requirementAndList, &envoy_jwt.JwtRequirement{
+			RequiresType: &envoy_jwt.JwtRequirement_RequiresAny{
+				RequiresAny: &envoy_jwt.JwtRequirementOrList{
+					Requirements: []*envoy_jwt.JwtRequirement{
+						{
+							RequiresType: &envoy_jwt.JwtRequirement_ProviderName{
+								ProviderName: name,
+							},
+						},
+						{
+							RequiresType: &envoy_jwt.JwtRequirement_AllowMissing{
+								AllowMissing: &empty.Empty{},
+							},
+						},
+					},
+				},
 			},
 		})
 	}
 
-	// TODO(diemvu): change to AllowMissing requirement.
-	requirementOrList = append(requirementOrList, &envoy_jwt.JwtRequirement{
-		RequiresType: &envoy_jwt.JwtRequirement_AllowMissingOrFailed{
-			AllowMissingOrFailed: &empty.Empty{},
-		},
-	})
+	if len(requirementAndList) == 1 {
```

Comment:
Why do we need special handling in this case?

Reply:
Envoy validation requires the list have at least 2 items.
https://github.com/envoyproxy/envoy/blob/master/api/envoy/extensions/filters/http/jwt_authn/v3alpha/config.proto#L347

Reply:
Discussed offline, the current PR doesn't support multiple issuers at the same location. We need either 1) fix this in the PR or 2) make it very clear in the documentation that this is not supported.

Reply:
worth a documentation comment here? since it's just to accomdate envoy api validation requiremetns.

Reply:
I've changed the config to support same location case. Unfortunately, we need to use smt like
A or B or ((A or missing) and (B or missing)), due to the existing implementation.

Updated code, tests and documentation for this.